definitions/checks/puppet/warn_about_puppet_removal.rb

```diff
@@ -0,0 +1,33 @@
+module Checks::Puppet
+  class WarnAboutPuppetRemoval < ForemanMaintain::Check
+    metadata do
+      description 'Warn about Puppet content removal prior to 6.10 upgrade'
+      label :warn_before_puppet_removal
+      confine do
+        feature(:instance).downstream &&
+          feature(:instance).downstream.current_minor_version == '6.9'
+      end
+    end
+
+    def run
+      message = "Puppet repositories found!\n"\
+                "Upgrading to 6.10 will delete the puppet repositories\n"\
+                "if any of the following conditions are met,\n"\
+                "1. Any puppet repositories in Library lifecycle environment.\n"\
+                "2. Unpublished content view's puppet repositories.\n"\
+                "Note: Any puppet content that is in use by hosts will not be deleted\n"\
+                "and, hosts will continue to use the puppet modules after upgrade.\n\n"\
+                'Do you want to proceed?'
```

Comment:
'hosts will continue to use the puppet modules after upgrade but further management of puppet content must take place outside of the Satellite application. '  or something like that?